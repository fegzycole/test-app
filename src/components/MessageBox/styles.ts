import Box from '@mui/material/Box';
import { styled } from '@mui/system';

export const StyledBox = styled(Box, {
    name: 'StyledBox',
})`
    margin: 100px auto;
    padding: 50px;
    border: 1px solid #000;
`;
