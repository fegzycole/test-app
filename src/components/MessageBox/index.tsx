import React from 'react';
import Typography from '@mui/material/Typography';
import { StyledBox } from './styles';

export interface MessageBoxProps {
    text: string;
}

export const MessageBox: React.FC<MessageBoxProps> = ({ text }) => (
    <StyledBox>
        <Typography variant="h5">{text}</Typography>
    </StyledBox>
);

export default MessageBox;
