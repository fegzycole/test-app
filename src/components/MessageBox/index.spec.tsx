import { cleanup, render } from '@testing-library/react';
import MessageBox, { MessageBoxProps } from './index';

const props: MessageBoxProps = {
    text: 'Mock Text',
};

describe('<MessageBox />', () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup();
    });

    it('should render correctly', () => {
        const view = render(<MessageBox {...props} />);

        expect(view.container).toMatchSnapshot();
    });
});
