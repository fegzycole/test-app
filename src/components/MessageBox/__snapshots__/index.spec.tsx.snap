// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageBox /> should render correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-qqlyqx-StyledBox-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Mock Text
    </h5>
  </div>
</div>
`;
